R/validate_inputs.R

```diff
@@ -108,6 +108,11 @@ validate_inputs <- function(..., header = "Some inputs require attention") {
   lapply(vals, checkmate::assert_class, "InputValidator")
   checkmate::assert_string(header, null.ok = TRUE)
 
+  if (!all(vapply(vals, validator_enabled, logical(1L)))) {
+    warning("Some validators are disabled and will be omitted.", call. = TRUE)
```

Comment:
Should we use a  WARN from teal.logger here? an R warning may be a bit too much

Reply:
You probably should stick to just one logging channel. Either use the built-in R messages and warnings or use the logger solution.

Reply:
The reasoning is that logger outputs can be piped to a file without R warnings. If someone was to do it, would they miss this warning?

Reply:
`teal.logger` it is then. Any advice on how to write unit tests for it?

Reply:
I've checked this issue here and situation is not that simple for me, so let me know if you have some solution or a strong opinion about how to handle this situation:

It's not straightforward how to test modules which return warnings through the `logger`. `options(TEAL.LOG_LEVEL)` + `capture_output` in tests doesn't seem to be optimal solution. I've checked there is no function in the `logger` to "convert" logger-conditions into the exceptions. 

With @chlebowa we tried to find the way to temporary (for tests) turn logger entries into R errors, warnings, messages and it's also not so easy neither. Currently, it's possible to do access `level` and `msg` in [layout_teal_glue_generator](https://github.com/insightsengineering/teal.logger/blob/b2108770072f8def9d5e372cd306ba9e51a062ae/R/register_logger.R#L106) via `register_logger` which would have to be called once again especially for tests.

Alternative solution (not for today) is to use `logger::log_errors`, `logger::log_warnings` and change all `logger::log_warn` in teal packages to generic `warning`. How would it work then? - `log_errors` uses `withGlobalCallingHandlers` and intercept errors (their messages) and append them into logger. Our modules then can work normally with `warning`, `stop` and they could be appended into logger if one adds `logger::log_errors` on top of the `app.R` file.

Reply:
As this is not a trivial issue, I have modified the tests to test console output as a temporary measure.